libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -675,144 +693,193 @@ open class LoginActivity : AppCompatActivity(), OAuthWebviewHelperEvents {
-     * @param loginQrCodeContent The login QR code content string
-     * @return UiBridgeApiParameters: The UI Bridge API parameters or null if the QR code
-     * content cannot provide them for any reason
-     */
-    private fun uiBridgeApiParametersFromLoginQrCodeContent(
-        loginQrCodeContent: String?
-    ) = loginQrCodeContent?.let { loginQrCodeContentUnwrapped ->
-        uiBridgeApiJsonFromQrCodeContent(loginQrCodeContentUnwrapped)?.let { uiBridgeApiJson ->
-            uiBridgeApiParametersFromUiBridgeApiJson(uiBridgeApiJson)
-        }
-    }
-
-    /**
-     * Parses UI Bridge API parameters JSON from the provided string, which may
-     * be formatted to match either QR code content provided by the intent or
-     * the app's QR code library.
+     * Automatically log in using a QR code login URL and Salesforce Identity API UI Bridge.
      *
-     * 1. From intent (external QR reader): ?bridgeJson={...}
-     * 2. From the app's QR reader: ?bridgeJson=%7B...%7D
+     * This method is the intended entry point for login using the reference QR Code Login URL and
+     * JSON format.  It will parse the UI Bridge parameters from the login QR code URL and call
+     * [LoginActivity.loginWithFrontdoorBridgeUrl].  However, the URL and JSON structure in the
+     * reference implementation is not required.  An app may use a custom structure so long as UI
+     * Bridge front door URL and optional PKCE code verifier are provided to
+     * [LoginActivity.loginWithFrontdoorBridgeUrl].
      *
-     * @param qrCodeContent The QR code content string
-     * @return String: The UI Bridge API parameter JSON or null if the string
-     * cannot provide the JSON for any reason
-     */
-    private fun uiBridgeApiJsonFromQrCodeContent(
-        qrCodeContent: String
-    ) = qrCodeBridgeJsonRegexExternal.find(qrCodeContent)?.groups?.get(1)?.value
-        ?: qrCodeBridgeJsonRegexInternal.find(qrCodeContent)?.groups?.get(1)?.value?.let {
-            URLDecoder.decode(it, "UTF-8")
-        }
-
-    /**
-     * Creates UI Bridge API parameters from the provided JSON string.
-     * @param uiBridgeApiParameterJsonString The UI Bridge API parameters JSON
-     * string
-     * @return The UI Bridge API parameters
+     * @param qrCodeLoginUrl The QR code login URL
+     * @return Boolean true if a log in attempt is possible using the provided QR code log in URL,
+     * false otherwise
      */
-    private fun uiBridgeApiParametersFromUiBridgeApiJson(
-        uiBridgeApiParameterJsonString: String
-    ) = JSONObject(uiBridgeApiParameterJsonString).let { uiBridgeApiParameterJson ->
-        UiBridgeApiParameters(
-            uiBridgeApiParameterJson.getString(FRONTDOOR_BRIDGE_URL_KEY),
-            when (uiBridgeApiParameterJson.has(PKCE_CODE_VERIFIER_KEY)) {
-                true -> uiBridgeApiParameterJson.optString(PKCE_CODE_VERIFIER_KEY)
-                else -> null
-            }
+    fun loginWithFrontdoorBridgeUrlFromQrCode(
```

Comment:
This is our entry point to MSDK's front door bridge URL login logic when the app does choose to use our reference implementation's APEX, Visualforce Page and the associated login QR code URL/JSON format.   The app just passes the URL from the QR code to MSDK and magic ensues.